pepper-apis/ddp-common/src/main/java/org/broadinstitute/ddp/appengine/spark/SparkBootUtil.java

```diff
@@ -89,10 +93,17 @@ public static void startSparkServer(AppEngineShutdown stopRouteCallback, Config
         });
 
         if (stopRouteCallback != null) {
-            Runtime.getRuntime().addShutdownHook(new Thread(() -> stopRouteCallback.onTerminate()));
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                isShuttingDown = true;
```

Comment:
question: Is there a case where this needs to go back to false, for example in case of an error?

Reply:
I'm thinking that this is really a one-way thing: once the signal to shutdown is received, the shutdown is going to happen, regardless of errors.  Check out [this doc](https://cloud.google.com/appengine/docs/standard/how-instances-are-managed#shutdown) that describes the shutdown process.